ext/nnstreamer/tensor_filter/tensor_filter_snpe.cc

```diff
@@ -0,0 +1,296 @@
+  setTensorProp (outputInfo, output_tensor_map);
+
+  output_tensor_map.clear ();
+
+  empty_model = false;
+}
+
+void snpe_subplugin::invoke (const GstTensorMemory *input, GstTensorMemory *output)
+{
+  assert (!empty_model);
+  assert (snpe);
+
+  /* Configure inputs */
+  for (unsigned int i = 0; i < inputInfo.num_tensors; ++i) { 
+    float *finput = (float *) input[i].data;
+    size_t fsize =  input_tensors[i].get ()->getSize ();
+    std::copy (finput, finput + fsize, input_tensors[i].get ()->begin ());
+  }
+
+  output_tensor_map.clear ();
+  snpe->execute (input_tensor_map, output_tensor_map);
+
+  for (unsigned int i = 0; i < outputInfo.num_tensors; ++i) {
+    zdl::DlSystem::ITensor *output_tensor = 
+        output_tensor_map.getTensor (output_tensor_map.getTensorNames ().at (i));
+    float *foutput = (float *) output[i].data;
+    std::copy (output_tensor->cbegin (), output_tensor->cend (), foutput);
+  }
+
+}
+
+void snpe_subplugin::getFrameworkInfo (GstTensorFilterFrameworkInfo &info)
+{
+  info.name = name;
+  info.allow_in_place = 0;
+  info.allocate_in_invoke = 0;
+  info.run_without_model = 0;
+  info.verify_model_path = 1;
+}
+
+int snpe_subplugin::getModelInfo (model_info_ops ops,
+    GstTensorsInfo &in_info, GstTensorsInfo &out_info)
+{
+  if (ops == GET_IN_OUT_INFO) {
+    gst_tensors_info_copy (std::addressof (in_info),
+        std::addressof (inputInfo));
+    gst_tensors_info_copy (std::addressof (out_info),
+        std::addressof (outputInfo));
+    return 0;
+  }
+
+  return -ENOENT;
+}
+
+int snpe_subplugin::eventHandler (event_ops ops,
+    GstTensorFilterFrameworkEventData &data)
+{
+  return -ENOENT;
+}
+
+void snpe_subplugin::setTensorProp (GstTensorsInfo & tensor_meta,
+    zdl::DlSystem::TensorMap & tensor_map)
+{
+  tensor_meta.num_tensors = tensor_map.size ();
+  for (unsigned int i = 0; i < tensor_map.size (); ++i) {
+    tensor_meta.info[i].name = g_strdup (tensor_map.getTensorNames ().at (i));
+    tensor_meta.info[i].type = _NNS_FLOAT32;
+
+    unsigned int rank = 
+      tensor_map.getTensor (tensor_meta.info[i].name)->getShape ().rank ();
+    for (unsigned int j = 0; j < rank; ++j) {
+      tensor_meta.info[i].dimension[j] = 
+        tensor_map.getTensor (tensor_meta.info[i].name)->getShape () [rank - j - 1];
+    }
+    for (unsigned int j = rank; j < NNS_TENSOR_RANK_LIMIT; ++j) {
+      tensor_meta.info[i].dimension[j] = 1;
+    }
+  }
+}
+
+snpe_subplugin *snpe_subplugin::registeredRepresentation = nullptr;
+
+/** @brief Initialize this object for tensor_filter subplugin runtime register */
+void snpe_subplugin::init_filter_snpe (void)
+{
+  registeredRepresentation =
+      tensor_filter_subplugin::register_subplugin<snpe_subplugin> ();
+}
+
+/** @brief Destruct the subplugin */
+void snpe_subplugin::fini_filter_snpe (void)
+{
+  assert (registeredRepresentation != nullptr);
+  tensor_filter_subplugin::unregister_subplugin (registeredRepresentation);
+}
+
+extern "C" {
```

Comment:
Why this is required?

Reply:
This statement is for nnstreamer-native-api

Reply:
You already declare `extern "C"` in line 41. So, I think this is not necessary.

Reply:
Thanks! I think you're right. I'll modify this part.